vm/devices/storage/disk_backend_resources/src/lib.rs

```diff
@@ -121,3 +100,52 @@ pub enum BlobDiskFormat {
     /// A fixed VHD1, with a VHD footer specifying disk metadata.
     FixedVhd1,
 }
+
+/// Handle for a disk that is backed by one or more layers.
+#[derive(MeshPayload)]
+pub struct LayeredDiskHandle {
+    /// The layers that make up the disk. The first layer is the top-most layer.
+    pub layers: Vec<DiskLayerDescription>,
+}
+
+impl LayeredDiskHandle {
+    /// Create a new layered disk handle with a single layer.
+    pub fn single_layer(layer: impl IntoResource<DiskLayerHandleKind>) -> Self {
+        Self {
+            layers: vec![layer.into_resource().into()],
+        }
+    }
+}
+
+impl ResourceId<DiskHandleKind> for LayeredDiskHandle {
+    const ID: &'static str = "layered";
+}
+
+/// Description of a disk layer.
+#[derive(MeshPayload)]
+pub struct DiskLayerDescription {
+    /// The layer resource.
+    pub layer: Resource<DiskLayerHandleKind>,
+    /// If true, reads that miss this layer are written back to this layer.
+    pub read_cache: bool,
+    /// If true, writes are written both to this layer and the next one.
+    pub write_through: bool,
+}
+
+impl From<Resource<DiskLayerHandleKind>> for DiskLayerDescription {
```

Comment:
should this impl, and the DiskLayerHandle impl, move into `mod layer`?

Reply:
I don't have any strong preferences here. Do you want all layer-related types and functionality in `layer`? Or just ones related to `DiskLayerHandleKind`?

Reply:
Just the `DiskLayerHandleKind`

Reply:
it seems reasonable to leave the layered-disk handle at the top level, alongside the other top-level disk kinds

Reply:
Oh, I see, I missed `DiskLayerHandle`.